src/Shared/JsonSchemaExporter/JsonSchemaExporter.cs

```diff
@@ -660,23 +660,14 @@ private static JsonSchema GetSchemaForNumericType(JsonSchemaType schemaType, Jso
         return new JsonSchema { Type = schemaType, Pattern = pattern };
     }
 
-    // Uses reflection to determine the element type of an enumerable or dictionary type
-    // Workaround for https://github.com/dotnet/runtime/issues/77306#issuecomment-2007887560
-    private static Type GetElementType(JsonTypeInfo typeInfo)
-    {
-        Debug.Assert(typeInfo.Kind is JsonTypeInfoKind.Enumerable or JsonTypeInfoKind.Dictionary, "TypeInfo must be of collection type");
-        _elementTypeProperty ??= typeof(JsonTypeInfo).GetProperty("ElementType", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
-        return (Type)_elementTypeProperty?.GetValue(typeInfo)!;
-    }
-
-    private static PropertyInfo? _elementTypeProperty;
-
     // The .NET 8 source generator doesn't populate attribute providers for properties
     // cf. https://github.com/dotnet/runtime/issues/100095
     // Work around the issue by running a query for the relevant MemberInfo using the internal MemberName property
     // https://github.com/dotnet/runtime/blob/de774ff9ee1a2c06663ab35be34b755cd8d29731/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs#L206
-    [RequiresUnreferencedCode(RequiresUnreferencedCodeMessage)]
-    private static ICustomAttributeProvider? ResolveAttributeProvider(Type? declaringType, JsonPropertyInfo? propertyInfo)
+    private static ICustomAttributeProvider? ResolveAttributeProvider(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
```

Comment:
The particular `GetMember` call in line 687 requires `All`.

Reply:
![image](https://github.com/user-attachments/assets/85e73dd9-210c-4c63-8875-e13d4bae69a5)

Reply:
It doesn't matter much in this context, the method is always fed a `declaringType` parameter that is statically unknown. I'm just using `DynamicallyAccessedMembers` to refine the annotation of the particular method.

Reply:
@MichalStrehovsky @agocke - should the analyzer be smarter here when calling `GetMember` with a hard-coded binding flags? or would the recommendation be to make 2 calls - one to GetProperty and one to GetField?

If we want to limit usages of `DynamicallyAccessedMemberTypes.All`, `Type.GetMember` seems to be a problem.

Reply:
* I'm working on #88512 as I'm typing this. We can likely relax the static annotation on `GetMember` to something less crazy when that's available.
* We had a contribution to improve analysis of GetMember here: #94879. This would intrinsically recognize things better, probably would allow the combination in the above screenshot. It didn't merge (cc @sbomer).
* Given the current state of things, doing two calls, one to GetProperty, the other to GetField is probably the best.

Reply:
(And yes, we should avoid .All at all reasonable costs. Extra line or two of code is a good tradeoff to get rid of a `.All` use.)

Reply:
> Given the current state of things, doing two calls, one to GetProperty, the other to GetField is probably the best.

The problem is we don't really know if the underlying member is a property or field, so this would require chaining one call after the other. Is it really worth the duplication?